### DISCUSSION TITLE
.NET 5

### DISCUSSION THREAD

**Discussion by sbwalker** [oqtane/oqtane.framework#787] [General]
So I migrated Oqtane to .NET 5 RC1 locally. The migration was fairly straightforward. I have tested and confirmed that it runs on both Server and WebAssembly. I was also able to confirm that modules/themes which were created on .NET Core 3.1 will still run in Oqtane on .NET 5 without any changes or recompilation required. The only issue that I have found so far is that .NET 5 creates a new runtime under the /bin folder ( ie. bin\Debug\net5.0 ). This is expected, however any external Oqtane module or theme DLLs will still be located in the old bin\Debug\netcoreapp3.1 folder - so they will need to be manually copied to the new location in order for Oqtane to be aware of them.

**Comment by sbwalker** [comment]
Note that the issue I described above related to runtime folders only exists in the development environment. The runtime environment does not use this same folder structure so the DLLs will be located in the correct location.

**Reply by hishamco** [reply]
> I described above related to runtime folders only exists in the development environment

How the app can target different frameworks?

**Reply by sbwalker** [reply]
Oqtane cannot target multiple frameworks - it targets a specific framework. Currently it targets .NET 3.1. In V2 it is going to target .NET 5.
